test/spec_response.rb

```diff
@@ -451,24 +473,12 @@ def object_with_each.each
     res.body.must_be :closed?
     b.wont_equal res.body
 
-    res.body = StringIO.new
-    res.status = 205
-    _, _, b = res.finish
-    res.body.wont_be :closed?
-    b.wont_equal res.body
-
     res.body = StringIO.new
     res.status = 304
     _, _, b = res.finish
     res.body.must_be :closed?
     b.wont_equal res.body
   end
-
-  it "wraps the body from #to_ary to prevent infinite loops" do
-    res = Rack::Response.new
-    res.finish.last.wont_respond_to(:to_ary)
-    lambda { res.finish.last.to_ary }.must_raise NoMethodError
-  end
```

Comment:
@tenderlove does this concern you too? Can you explain what kind of infinite loops it's trying to prevent?

Reply:
No, this test makes no sense to me.  Let's ✂️

Reply:
It looks like this test was introduced in 5b251a986962f3577e14c01a698fab9dd21383ba to fix #419. It seems like we shouldn't delete it, but maybe we should call `flatten` rather than checking for `to_ary`?

It seems OK for something to respond to `to_ary`, but it doesn't seem OK to produce a circular data structure.

Reply:
Fixed in https://github.com/rack/rack/pull/1453